### PR TITLE
Don't ignore the NotFoundErr when handling the status hook of tasks to avoid the status out of sync

### LINKED ISSUE #14016: One of  GC schedule job is in running status all the time
image

### PR DESCRIPTION
**Pull request by ywk253100**
Don't ignore the NotFoundErr when handling the status hook of tasks to avoid the status out of sync
Fixes #14016

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/controller/replication/controller.go [hunk 1]
```diff
@@ -95,6 +95,16 @@ func (c *controller) Start(ctx context.Context, policy *model.Policy, resource *
 		// may be submitted already when the process starts, so create a new context
 		// with orm populated
 		ctxx := orm.NewContext(context.Background(), c.ormCreator.Create())
+
+		// as we start a new transaction in the goroutine, the execution record may not
+		// be inserted yet, wait until it is ready before continue
+		if err := lib.RetryUntil(func() error {
```
Comment (steven-zou) [inline src/controller/replication/controller.go:101]: RetryUntil(func() error, duration)?

Comment (steven-zou) [reply src/controller/replication/controller.go:101]: The name is a little confusing now.

File: src/jobservice/common/rds/scripts.go [hunk 2]
```diff
@@ -37,19 +37,17 @@ end
...
   local sCode = stCode(status)
   local aCode = stCode(ARGV[1])
   local aRev = tonumber(ARGV[2]) or 0
   local aCheckInT = tonumber(ARGV[3]) or 0
-
   if revision < aRev or 
-    ( revision == aRev and sCode < aCode ) or
-    ( revision == aRev and sCode == aCode and (not checkInT or checkInT < aCheckInT))
+    ( revision == aRev and sCode <= aCode ) or
+    ( revision == aRev and aCheckInT)
```
Comment (steven-zou) [inline src/jobservice/common/rds/scripts.go:47]:
```suggestion
    ( revision == aRev and aCheckInT ~= 0)
```

File: src/server/handler/job_status_hook.go [hunk 3]
```diff
@@ -45,11 +43,17 @@ func (j *jobStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		// Do not ignore the NotFoundErr here to make jobservice resend the status hook
+		// again to avoid the status lost
+		/*
+			// ignore the not found error to avoid the jobservice re-sending the hook
+			if errors.IsNotFoundErr(err) {
+				log.Warningf("got not found error: %v, ignore it to avoid subsequent retrying webhooks from jobservice", err)
+				return
+			}
+		*/
 		libhttp.SendError(w, err)
```
Comment (steven-zou) [inline src/server/handler/job_status_hook.go:57]: What about other none-fixable errors?

**Review by steven-zou — COMMENTED** [review]
check my comments

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by heww — APPROVED** [review]

**Pull request merged**